src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -169,14 +189,30 @@ public void Start(object? parameter)
                 }
 
                 startHelper._startArg = parameter;
-                startHelper._executionContext = ExecutionContext.Capture();
+                startHelper._executionContext = captureContext ? ExecutionContext.Capture() : null;
             }
 
             StartCore();
         }
 
+        /// <summary>Causes the operating system to change the state of the current instance to <see cref="ThreadState.Running"/>.</summary>
+        /// <exception cref="ThreadStateException">The thread has already been started.</exception>
+        /// <exception cref="OutOfMemoryException">There is not enough memory available to start this thread.</exception>
+        [UnsupportedOSPlatform("browser")]
+        public void Start() => Start(captureContext: true);
+
+        /// <summary>Causes the operating system to change the state of the current instance to <see cref="ThreadState.Running"/>.</summary>
+        /// <exception cref="ThreadStateException">The thread has already been started.</exception>
+        /// <exception cref="OutOfMemoryException">There is not enough memory available to start this thread.</exception>
+        /// <remarks>
+        /// Unlike <see cref="Start"/>, which captures the current <see cref="ExecutionContext"/> and uses that context to invoke the thread's delegate,
+        /// <see cref="UnsafeStart"/> explicitly avoids capturing the current context and flowing it to the invocation.
+        /// </remarks>
+        [UnsupportedOSPlatform("browser")]
+        public void UnsafeStart() => Start(captureContext: false);
+
         [UnsupportedOSPlatform("browser")]
```

Comment:
Are these attributes needed on the private implementation details?

Reply:
@buyaa-n?

Reply:
I have asked because many of the internal implementation details are missing these annotations currently, e.g. `Thread.StartCore()`. It suggests that these annotations are not needed on internal implementation details.

Reply:
Yeah, I just tried without it and there aren't any warnings. Removed.  I think it would only be needed if something statically reachable was annotated incompatibly.